### PR TITLE
[enhence] Make step_input of dynamic_rnn support custom lod level.

### PR DESCRIPTION
**Pull request by wanghaoshuang**

### CODE REVIEW

File: python/paddle/fluid/layers/control_flow.py [hunk 1]
```diff
@@ -1448,12 +1448,13 @@ def __init__(self, name=None):
...
         self.mem_link = []
 
-    def step_input(self, x):
+    def step_input(self, x, level=0):
         """
         Mark a sequence as a dynamic RNN input.
 
         Args:
             x(Variable): The input sequence.
+            level(int): The level of lod used to split steps.
```
Comment (shanyi15) [inline python/paddle/fluid/layers/control_flow.py]: level的默认值要不要在参数说明里专门说下？

Comment (wanghaoshuang) [reply python/paddle/fluid/layers/control_flow.py]: Fixed. 多谢提醒。

File: python/paddle/fluid/layers/control_flow.py [hunk 2]
```diff
@@ -1448,12 +1448,13 @@ def __init__(self, name=None):
...
         self.mem_link = []
 
-    def step_input(self, x):
+    def step_input(self, x, level=0):
         """
         Mark a sequence as a dynamic RNN input.
 
         Args:
             x(Variable): The input sequence.
+            level(int): The level of lod used to split steps. Default: 1.
```
Comment (panyx0718) [inline python/paddle/fluid/layers/control_flow.py]: default is 1 or 0?

Comment (wanghaoshuang) [reply python/paddle/fluid/layers/control_flow.py]: Sorry. It should be 0. Fixed.

**Comment by shanyi15** [comment]
注释辛苦和https://github.com/PaddlePaddle/Paddle/pull/15431 一起预览下，谢谢！

**Comment by wanghaoshuang** [comment]
image

**Comment by haowang101779990** [comment]
API preview LGTM

**Pull request merged**